docs/content/DimensionSpecs.md

```diff
@@ -154,3 +154,195 @@ Example for the `__time` dimension:
+|`valueColumn`|The name of the column containing the value|no|The second column|
+
+*example input*
+```
+bar,something,foo
+bat,something2,baz
+truck,something3,buck
+```
+
+*example parseSpec*
+```json
+"parseSpec": {
+  "type": "csv",
+  "columns": ["value","somethingElse","key"],
+  "keyColumn": "key",
+  "valueColumn": "value"
+}
+```
+
+##### tsv ParseSpec
+
+|Parameter|Description|Required|Default|
+|---------|-----------|--------|-------|
+|`columns`|The list of columns in the csv file|yes|`null`|
+|`keyColumn`|The name of the column containing the key|no|The first column|
+|`valueColumn`|The name of the column containing the value|no|The second column|
+|`delimiter`|The delimiter in the file|no|tab (`\t`)|
+
+
+*example input*
+```
+bar|something,1|foo
+bat|something,2|baz
+truck|something,3|buck
+```
+
+*example parseSpec*
+```json
+"parseSpec": {
+  "type": "tsv",
+  "columns": ["value","somethingElse","key"],
+  "keyColumn": "key",
+  "valueColumn": "value",
+  "delimiter": "|"
+}
+```
+
+##### customJson ParseSpec
+
+|Parameter|Description|Required|Default|
+|---------|-----------|--------|-------|
+|`keyFieldName`|The field name of the key|yes|null|
+|`valueFieldName`|The field name of the value|yes|null|
+
+*example input*
+```json
+{"key": "foo", "value": "bar", "somethingElse" : "something"}
+{"key": "baz", "value": "bat", "somethingElse" : "something"}
+{"key": "buck", "somethingElse": "something", "value": "truck"}
+```
+
+*example parseSpec*
+```json
+"parseSpec": {
+  "type": "customJson",
+  "keyFieldName": "key",
+  "valueFieldName": "value"
+}
+```
+
+
+##### simpleJson ParseSpec
+The `simpleJson` parseSpec does not take any parameters. It is simply a line delimited json file where the field is the key, and the field's value is the value.
+
+*example input*
+ 
+```json
+{"foo": "bar"}
+{"baz": "bat"}
+{"buck": "truck"}
+```
+
+*example parseSpec*
+```json
+"parseSpec":{
+  "type": "simpleJson"
+}
+```
+
+
+#### JDBC namespace update
+The JDBC namespaces will pull from a database cache. If the `tsColumn` is set it must be able to accept comparisons in the format `'2015-01-01 00:00:00'`. For example, the following must be valid sql for the table `SELECT * FROM some_namespace_table WHERE timestamp_column >  '2015-01-01 00:00:00'`. If `tsColumn` is set, the caching service will attempt to only pull values that were written *after* the last sync. If `tsColumn` is not set, the entire table is pulled every time.
+
+```json
+{
+  "namespace":{
+    "namespace":"some_namespace",
+    "connectorConfig":{
+      "createTables":true,
+      "connectURI":"jdbc:mysql://localhost:3306/druid",
+      "user":"druid",
+      "password":"diurd"
+    },
+    "table":"some_namespace_table",
+    "keyColumn":"the_old_dim_value",
+    "valueColumn":"the_new_dim_value",
+    "tsColumn":"timestamp_column"
+  },
+  "updateMs":600000
+}
+```
+
+### Kafka namespace update (Experimental)
```

Comment:
For extensions that are support as part of the main distribution (i.e. anything in this repo), it makes sense to document them as part of the existing documentation. It would make sense however to create a separate section for individual extensions, in order to avoid confusion as to what extensions are required for a given feature.